src/vs/workbench/parts/scm/electron-browser/scmViewlet.ts

```diff
@@ -958,11 +964,14 @@ export class RepositoryPanel extends ViewletPanel {
 	}
 
 	private updateInputBox(): void {
-		if (typeof this.repository.provider.commitTemplate === 'undefined') {
+		const commitTemplate = this.repository.provider.commitTemplate;
+
+		if (typeof commitTemplate !== 'undefined' && commitTemplate !== '') {
+			this.inputBox.value = commitTemplate;
```

Comment:
So if you have a commitTemplate, it's impossible to restore the recovered message? I think it should be opposite, if there is a saved message then it should be used instead of the template.